app/src/main/res/values/strings.xml

```diff
@@ -67,6 +67,10 @@
   <string name="play_pause_audio">Play/Pause Audio</string>
   <string name="help">Help</string>
   <string name="preferences">Preferences</string>
+  <string name="fraction_error_invalid_chars">Please only use numerical digits, spaces or forward slashes (/)</string>
+  <string name="fraction_error_valid">valid</string>
```

Comment:
It seems a bit odd to have a string for this--can we omit it by making the string res ID optional in the enum?

Reply:
Removed. https://gist.github.com/nikitamarysolomanpvt/8fed61722b472df342dac71680296056 is this what you want me to do?

Reply:
More or less, though https://gist.github.com/BenHenning/b418411a609ca0374516395bcd6e28b8 might simplify it a bit (it propagates the null upward instead of trying to handle it here--we ideally just don't want to set the text error if it's null, or decide to use an empty string higher up).

Reply:
Done.